libs/concurrency/include/hpx/concurrency/register_locks.hpp

```diff
@@ -26,7 +26,7 @@ namespace hpx { namespace util {
     template <typename Lock, typename Enable = void>
     struct ignore_while_checking;
 
-#if defined(HPX_HAVE_VERIFY_LOCKS) || defined(HPX_EXPORTS)
+#if defined(HPX_HAVE_VERIFY_LOCKS)
```

Comment:
What is the rationale for this change?

Reply:
Me being clueless... We had problems getting this to link correctly and tried a few things. I think @auraner will change it to be `|| defined HPX_MODULE_EXPORTS`.

Part of the problems came from the `HPX_DEBUG` here: https://github.com/STEllAR-GROUP/hpx/blob/58bc53642b4e13601e071f62bcbc1d3716140b7c/examples/hello_world_component/CMakeLists.txt#L30. Would you happen to know what the idea behind it is? Is it explicitly there to check that we are (more or less) ABI compatible between debug and release?

Reply:
Yes, this was an attempt to keep things ABI compatible between Release/Debug - and yes I know we don't do this consistently...

Reply:
Thanks, makes sense. We'll try not to make things at least worse then, in case we still decide to do this consistenly across the codebase at some point.